flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/logical/SymbolType.java

```diff
@@ -21,52 +21,57 @@
 import org.apache.flink.annotation.PublicEvolving;
 import org.apache.flink.table.api.TableException;
 import org.apache.flink.table.expressions.TableSymbol;
-import org.apache.flink.util.Preconditions;
 
 import java.util.Collections;
 import java.util.List;
-import java.util.Objects;
 
 /**
  * Logical type for representing symbol values. The symbol type is an extension to the SQL standard
  * and only serves as a helper type within the expression stack.
  *
- * <p>A symbol type only accepts conversions from and to its enum class.
+ * <p>A symbol type accepts conversions from and to {@link Enum}. But note that this is not an enum
+ * type for users.
  *
  * <p>This type has no serializable string representation.
  *
- * @param <T> table symbol
+ * @param <T> Legacy generic that will be dropped in the next major version. If we dropped it
+ *     earlier, we would break {@link LogicalTypeVisitor} implementation.
  */
+@SuppressWarnings("unused")
```

Comment:
leftover?

Reply:
this is used for the argument of the constructor

Reply:
Then maybe only move it there on top of the ctor?

Reply:
or like this: `public SymbolType(@SuppressWarnings("unused") Class<T> clazz) {` ?